### PR TITLE
Internal: integration tests

### PR DESCRIPTION
**Pull request by odesenfans**
Added a new crate dedicated to integration tests for the RPC client +
server. Integration tests spawn the server as an async task, bind it to
a Unix socket and connect a client to it.

This PR required the following refactorings:
* The `stone-prover` build.rs script now outputs the prover and verifier
  at the root of the target directory to make them available to all the
  crates in the workspace.
* The `madara-prover-rpc-server` crate now provides both a library and a
  binary. The lib makes it easier to spawn the server programmatically
  from another crate.
* Moved all test case files from `test-fixtures` crate to the new `test-cases`
  crate.
* Moved the prover fixtures to `test-fixtures`.

### CODE REVIEW

File: madara-prover-rpc-server/Cargo.toml [hunk 1]
```diff
@@ -14,7 +20,7 @@ thiserror = { workspace = true }
 tokio = { workspace = true }
 tonic = { workspace = true }
 serde_json = { workspace = true }
-tokio-stream = "0.1.14"
+tokio-stream = { version = "0.1.14", features = ["net"] }
```
Comment (HermanObst) [inline madara-prover-rpc-server/Cargo.toml:23]: Is this for networking handling?

**Review by HermanObst — APPROVED** [review]
It all seems good :) 
I would like if we can go through how the different test cases are being passed to the `test_integration`

**Comment by HermanObst** [comment]
Other important thing that is missing for the Milestone (because is what they actually want to use the API) is to provide a Client method that allows to:
- Pass the compiled program and its inputs
- Directly return the proof

The method in prover.proto would look like:
service Prover {
    rpc ExecuteProve(ExecutionRequest) returns (stream ProverResponse);
}

For this we will need to have in `madara-prover-rpc-server/src/lib.rs` a method that combines:
`async fn execute` and pass the `execution_result` to `async fn prove`

**Comment by odesenfans** [comment]
@HermanObst as mentioned, the `ExecuteAndProve` endpoint is coming with another PR that I'll create as soon as I have a functional E2E test.

**Pull request merged**